x-pack/platform/plugins/shared/cases/public/components/case_view/case_view_page.tsx

```diff
@@ -83,9 +84,12 @@ export const CaseViewPage = React.memo<CaseViewPageProps>(
       [onUpdateField]
     );
 
+    const location = useLocation();
+
     return (
       <>
         <HeaderPage
+          key={location.key}
```

Comment:
Could we go even further on the tree like setting the key on `EditableTitle`? Also, what about using the case ID (`caseData.id`) instead of the `location.key`?